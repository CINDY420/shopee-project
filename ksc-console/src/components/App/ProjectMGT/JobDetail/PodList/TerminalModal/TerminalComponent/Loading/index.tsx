import * as React from 'react'
import {
  Root,
  LoadingImage,
  LoadingText,
} from 'components/App/ProjectMGT/JobDetail/PodList/TerminalModal/TerminalComponent/Loading/style'

const Loading: React.FC = () => (
  <Root>
    <LoadingImage />
    <LoadingText>loading...</LoadingText>
  </Root>
)

export default Loading
