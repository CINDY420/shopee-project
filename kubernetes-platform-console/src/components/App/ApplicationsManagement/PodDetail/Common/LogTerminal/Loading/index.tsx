import * as React from 'react'

import {
  Root,
  LoadingImage,
  LoadingText
} from 'components/App/ApplicationsManagement/PodDetail/Common/LogTerminal/Loading/style'

const Loading: React.FC = () => {
  return (
    <Root>
      <LoadingImage />
      <LoadingText>Container logging into...</LoadingText>
    </Root>
  )
}

export default Loading
