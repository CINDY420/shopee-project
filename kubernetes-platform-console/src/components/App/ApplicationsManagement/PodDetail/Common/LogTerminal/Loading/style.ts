import styled from 'styled-components'

import ContainerLogin from 'assets/container_login.svg'

export const Root = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 8em;
  width: 100%;
  height: 100%;
  background-color: #000000;
  position: absolute;
  top: 0;
  z-index: 1000;
`

export const LoadingImage = styled.div`
  width: 15em;
  height: 15em;
  background-image: url(${ContainerLogin});
  background-repeat: no-repeat;
`

export const LoadingText = styled.span`
  position: relative;
  top: -2.5em;
  font-size: 1.5em;
  color: #00b9c8;
`
