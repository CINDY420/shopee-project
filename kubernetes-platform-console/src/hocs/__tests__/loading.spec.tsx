import * as React from 'react'
import { render } from '@testing-library/react'
import Loading from '../loading'

describe('Loading render UI result', () => {
  it('should return StyledSpin when loading is true', () => {
    const text = 'loading is true'
    const LoadingElement = Loading(() => <div>{text}</div>, true)
    const { container } = render(<LoadingElement />)

    expect(container).toMatchSnapshot()
  })

  it('should return DecoratedComponent when loading is false', () => {
    const text = 'loading is false'
    const LoadingElement = Loading(() => <div>{text}</div>, false)
    const { container } = render(<LoadingElement />)

    expect(container).toMatchSnapshot()
  })
})
