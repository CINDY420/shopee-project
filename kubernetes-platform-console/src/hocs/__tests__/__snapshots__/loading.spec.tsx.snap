// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading render UI result should return DecoratedComponent when loading is false 1`] = `
<div>
  <div>
    loading is false
  </div>
</div>
`;

exports[`Loading render UI result should return StyledSpin when loading is true 1`] = `
<div>
  <div
    class="ant-spin ant-spin-spinning sc-bdfBwQ eYtIxx"
  >
    <span
      aria-label="loading"
      class="anticon anticon-loading anticon-spin ant-spin-dot"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="loading"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 32 32"
        width="1em"
      >
        <path
          d="M19.02.8a15.4 15.4 0 017.94 4.24A15.45 15.45 0 0131.5 16a15.45 15.45 0 01-12.48 15.2 1.5 1.5 0 01-.58-2.94 12.39 12.39 0 006.4-3.42A12.45 12.45 0 0028.5 16 12.45 12.45 0 0018.44 3.74a1.5 1.5 0 01.38-2.97l.2.02z"
        />
      </svg>
    </span>
  </div>
</div>
`;
